tinygrad/tensor.py

```diff
@@ -205,16 +204,20 @@ def detach(self):
     return Tensor(self.data, device=self.device)
 
   # ***** non first class ops *****
-  
+
   def __getitem__(self, val):
     arg = []
-    for i,s in enumerate(val if type(val) in [list, tuple] else ([] if val is None else [val])):
-      if type(s) is int: arg.append((s, s+1))
-      else:
-        arg.append((s.start if s.start is not None else 0,
-          (s.stop if s.stop >=0 else self.shape[i]+s.stop) if s.stop is not None else self.shape[i]))
-      if type(s) is not int: assert s.step is None or s.step == 1
-    return self.slice(arg = arg+[(0,self.shape[i]) for i in range(len(arg), len(self.shape))])
+    if val is not None:
```

Comment:
I find the top one shorter and more readable.

Reply:
i agree that splitting up the list/tuple check was subjective and i can inline that again, but i would definitely keep the `is not None` check as it's own block since it doesn't need to make an empty list and an enumerate object in the case of None..

`for i,s in enumerate(val if type(val) in [list, tuple] else ([] if val is None else [val]))` this line really looks like something out of my first year of programming lol its so nasty